testsuite/MDAnalysisTests/utils/test_distances.py

```diff
@@ -33,9 +34,22 @@
 from MDAnalysis.lib import mdamath
 from MDAnalysisTests import parser_not_found
 
-class _TestDistanceArray(TestCase):
-    # override backend in test classes
-    backend = None
+@pytest.fixture()
+def ref_system():
+    box = np.array([1., 1., 2.], dtype=np.float32)
+    points = np.array(
+        [
+            [0, 0, 0], [1, 1, 2], [1, 0, 2],  # identical under PBC
+            [0.5, 0.5, 1.5],
+        ], dtype=np.float32)
+    ref = points[0:1]
+    conf = points[1:]
+
+    return box, points, ref, conf
+    
+
+@pytest.mark.parametrize('backend', ['serial', 'openmp'])
```

Comment:
Rather than having a base class, the subclassing it to create tests with a certain flavour, we can decorate the class like this to make it run all tests with each of the backends. ie
```python
class _BaseTestA(object):
    def test_a():
    
    def test_b()

class TestAFoo(_BaseTestA):
    flavour='vanilla'

class TestABar(_BaseTestA):
    flavour='chocolate'

    def test_extra_chocolate():
        pass
```


I prefer this as we end up with 1 class not 3, but it does mean we can't add tests for a specific flavour in the same way as we have elsewhere (`test_extra_chocolate` in the above example)